src/rpc/connectivity/cluster.cc

```diff
@@ -380,17 +388,21 @@ void connectivity_cluster_t::run_t::connect_to_peer(
     rate_control->unlock(1);
 }
 
-void connectivity_cluster_t::run_t::join_blocking(
+join_result_t connectivity_cluster_t::run_t::join_blocking(
         const peer_address_t peer,
         boost::optional<peer_id_t> expected_id,
+        boost::optional<server_id_t> expected_server_id,
         const int join_delay_secs,
         auto_drainer_t::lock_t drainer_lock) THROWS_NOTHING {
     drainer_lock.assert_is_holding(&parent->current_run->drainer);
     parent->assert_thread();
     {
         mutex_assertion_t::acq_t acq(&attempt_table_mutex);
         if (attempt_table.find(peer) != attempt_table.end()) {
-            return;
+            // NOTE: this is triggered all the time in the old code and ignored as what
```

Comment:
Doesn't this just mean that we have an active connection attempt to this peer already?
I think returning a TEMPORARY_ERROR is correct in that case. We don't know yet if the other join attempt is going to actually work out (plus even if it does, it might get interrupted later and we still need to try reconnecting using this address).